app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt

```diff
@@ -126,19 +130,28 @@ private fun Username(username: String) {
 @Composable
 private fun MessageContent(messageContent: MessageContent) {
     when (messageContent) {
-        is MessageContent.ImageMessage -> MessageImage(imageUrl = messageContent.imageUrl)
+        is MessageContent.ImageMessage -> MessageImage(rawImgData = messageContent.rawImgData, messageContent.width, messageContent.height)
         is MessageContent.TextMessage -> MessageBody(messageBody = messageContent.messageBody)
     }
 }
 
-//TODO: replace with actual imageUrl loading probably with: https://coil-kt.github.io/coil/compose/
 @Composable
-fun MessageImage(imageUrl: String = "") {
+fun MessageImage(rawImgData: ByteArray, realImgWidth: Int, realImgHeight: Int) {
```

Comment:
how about creating some sort of data class for example that would hold this values and also have transform method to do what you did here : 

```
    val width = MaterialTheme.wireDimensions.messageImageMaxWidth
    val height = width.value * realImgHeight.toFloat() / realImgWidth
    ```

Reply:
I created a wrapper class called `ImageMessageParams` containing these normalized width and height values